### ISSUE TITLE
Wrong use of transpose in np.array() in fortran

### ISSUE THREAD

**Issue by ohachim** [pyccel/pyccel#1241]
**Describe the bug**
Wrong behaviour when using steps in views with np.array() bug in fortran translation

**To Reproduce**
Provide code to reproduce the behavior:
```python
import numpy as np
if __name__ == '__main__':
    a = np.array([[ 1,  2,  4,  8, 16],
       [ 1,  2,  4,  8, 16],
       [ 1,  2,  4,  8, 16],
       [ 1,  2,  4,  8, 16],
       [ 1,  2,  4,  8, 16]], order="F")
    tmp = a[::2]
    b = np.array(tmp, order="F")
    print(b)
```

Provide the generated code, or the error message:
```c
program prog_prog_test

  use test_0001

  use, intrinsic :: ISO_C_Binding, only : i64 => C_INT64_T
  use, intrinsic :: ISO_FORTRAN_ENV, only : stdout => output_unit
  implicit none

  integer(i64), allocatable, target :: a(:,:)
  integer(i64), pointer :: tmp(:,:)
  integer(i64), allocatable :: b(:,:)
  integer(i64) :: i
  integer(i64) :: i_0001
  integer(i64) :: i_0002

  allocate(a(0:4_i64, 0:4_i64))
  a = transpose(reshape([[1_i64, 2_i64, 4_i64, 8_i64, 16_i64], [1_i64, &
        2_i64, 4_i64, 8_i64, 16_i64], [1_i64, 2_i64, 4_i64, 8_i64, &
        16_i64], [1_i64, 2_i64, 4_i64, 8_i64, 16_i64], [1_i64, 2_i64, &
        4_i64, 8_i64, 16_i64]], [5_i64, 5_i64]))
  tmp(0:, 0:) => a(::2_i64, :)
  allocate(b(0:size(tmp, 1_i64, i64) - 1_i64, 0:size(tmp, 2_i64, i64) - &
        1_i64))
  b = transpose(tmp)
  write(stdout, '(A)', advance="no") '['
  do i = 0_i64, size(b, 1_i64, i64) - 1_i64 - 1_i64, 1_i64
    write(stdout, '(A)', advance="no") '['
    do i_0001 = 0_i64, size(b, 2_i64, i64) - 1_i64 - 1_i64, 1_i64
      write(stdout, '(I0 A)', advance="no") b(i, i_0001) , ' '
    end do
    write(stdout, '(I0 A)', advance="no") b(i, size(b, 2_i64, i64) - &
          1_i64) , ']'
    write(stdout, '()', advance="yes") 
  end do
  write(stdout, '(A)', advance="no") '['
  do i_0002 = 0_i64, size(b, 2_i64, i64) - 1_i64 - 1_i64, 1_i64
    write(stdout, '(I0 A)', advance="no") b(size(b, 1_i64, i64) - 1_i64, &
          i_0002) , ' '
  end do
  write(stdout, '(I0 A)', advance="no") b(size(b, 1_i64, i64) - 1_i64, &
        size(b, 2_i64, i64) - 1_i64) , ']'
  write(stdout, '(A)', advance="no") ']'
  write(stdout, '()', advance="yes") 
  if (allocated(a)) then
    deallocate(a)
  end if
  if (allocated(b)) then
    deallocate(b)
  end if

end program prog_prog_test
```

**Expected behavior**
Should print
```
[[1 2 4 8 16]
[1 2 4 8 16]
[1 2 4 8 16]]
```
Instead prints:
```
[[16 129 16]
[16 1 1]
[16 2 2]
[16 4 4]
[16 8 8]]
```

**Language**
fortran

 **Additional info**
Doesn't happen in 1D arrays

**Comment by jalalium** [comment]
This issue is common to all implicit assignments to Fortran.
**Describe the bug**
Wrong behaviour when using steps in views with np.array() bug in fortran translation

**To Reproduce**
Provide code to reproduce the behavior:
```python
import numpy as np
if __name__ == '__main__':
    a = np.array([[ 1,  2,  4,  8, 16],
       [ 1,  2,  4,  8, 16],
       [ 1,  2,  4,  8, 16],
       [ 1,  2,  4,  8, 16],
       [ 1,  2,  4,  8, 16]], order="F")
    b = np.array(a, order="F")
    print(b)
```

Provide the generated code, or the error message:
```c
program prog_prog_test

  use test_0001

  use, intrinsic :: ISO_C_Binding, only : i64 => C_INT64_T
  use, intrinsic :: ISO_FORTRAN_ENV, only : stdout => output_unit
  implicit none

  integer(i64), allocatable :: a(:,:)
  integer(i64), allocatable :: b(:,:)
  integer(i64) :: i
  integer(i64) :: i_0001
  integer(i64) :: i_0002

  allocate(a(0:4_i64, 0:4_i64))
  a = transpose(reshape([[1_i64, 2_i64, 4_i64, 8_i64, 16_i64], [1_i64, &
        2_i64, 4_i64, 8_i64, 16_i64], [1_i64, 2_i64, 4_i64, 8_i64, &
        16_i64], [1_i64, 2_i64, 4_i64, 8_i64, 16_i64], [1_i64, 2_i64, &
        4_i64, 8_i64, 16_i64]], [5_i64, 5_i64]))
  allocate(b(0:4_i64, 0:4_i64))
  b = transpose(a)
  write(stdout, '(A)', advance="no") '['
  do i = 0_i64, 3_i64, 1_i64
    write(stdout, '(A)', advance="no") '['
    do i_0001 = 0_i64, 3_i64, 1_i64
      write(stdout, '(I0 A)', advance="no") b(i, i_0001) , ' '
    end do
    write(stdout, '(I0 A)', advance="no") b(i, 4_i64) , ']'
    write(stdout, '()', advance="yes") 
  end do
  write(stdout, '(A)', advance="no") '['
  do i_0002 = 0_i64, 3_i64, 1_i64
    write(stdout, '(I0 A)', advance="no") b(4_i64, i_0002) , ' '
  end do
  write(stdout, '(I0 A)', advance="no") b(4_i64, 4_i64) , ']'
  write(stdout, '(A)', advance="no") ']'
  write(stdout, '()', advance="yes") 
  if (allocated(a)) then
    deallocate(a)
  end if
  if (allocated(b)) then
    deallocate(b)
  end if

end program prog_prog_test
```

**Expected behavior**
Should print
```
[[ 1  2  4  8 16]
 [ 1  2  4  8 16]
 [ 1  2  4  8 16]
 [ 1  2  4  8 16]
 [ 1  2  4  8 16]]
```
Instead prints:
```
[[1 1 1 1 1]
[2 2 2 2 2]
[4 4 4 4 4]
[8 8 8 8 8]
[16 16 16 16 16]]
```
The issue is due to an extra `transpose` during assignment. Instead of adding a systematic `transpose` after assignment when we are dealing with `Fortran` Format, we should check if the transpose has already been performed for the symbol we are dealing with. I am not sure if we can fix this by creating an `AliasAssign` as it is the case with C.

**Comment by EmilyBourne** [comment]
Given the fix in #1256 am I right that this is a Duplicate of #1257 ? It has nothing to do with steps?

**Comment by yguclu** [comment]
> Given the fix in #1256 am I right that this is a Duplicate of #1257 ? It has nothing to do with steps?

Yes, I think so.
